specification/sbi_atee.adoc

```diff
@@ -13,6 +43,858 @@ running a VMM).
 The following operations must be implemented by the TSM ( _Formal input 
 parameters and return codes are TBD_ ).
 
+== TEE Extension (EID #0x41544545)
+
+=== Function: Create TVM (FID #0)
+[source, C]
+-----
+struct sbiret sbi_tee_create_tvm(unsigned long tvm_create_params,
+                                 unsigned long tvm_create_params_len);
+-----
+Creates a confidential TVM using the specified parameters. The tvm_create_parameters
+is a pointer to the `tvm_create_params` described below, and tvm_create_params_len is
+the size of the structure in bytes. 
+
+Callers of this API should first invoke `sbi_tee_tsm_get_info()` to obtain information
+about the parameters that should be used to populate `tvm_create_params`.
+
+[source, C]
+----
+struct tvm_create_params {
+    /* 
+    The base physical address of the 16KiB confidential memory region
+    that should be used for the TVM’s page directory. Must be 16KiB-aligned.
+    */
+    unsigned long tvm_page_directory_addr;
+    /* 
+    The base physical address of the confidential memory region to be used
+    to hold the TVM’s global state. Must be page-aligned and the number of
+    pages must be at least the value returned in tsm_info::vm_state_pages.
+    */
+    unsigned long tvm_state_addr;
+    /*
+    The maximum number of vCPUs that will be created for this TVM. Must be
+    less than or equal to the value returned in tsm_info::tvm_max_vcpus
+    */ 
+    unsigned long tvm_num_vcpus;
+    /* The base physical address of the confidential memory region to be used
+    to hold the TVM’s vCPU state. Must be page-aligned and must be at least
+    TsmInfo::tvm_bytes_per_vcpu * tvm_num_vcpus bytes in length, rounded up
+    to the nearest multiple of 4KiB */
+    unsigned long tvm_vcpu_addr;
+};
+----
+
+*Returns* the `tvm_guest_id` in sbiret.value on success. The `tvm_guest_id` can be used
+to uniquely reference the TVM in invocations of other sbi_tee functions.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_create_tvm_errors]
+.TEE Create TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The TVM with `tvm_guest_id` was successfully created.
+| SBI_ERR_INVALID_PARAM | The `tvm_create_params` or `tvm_create_params_len` parameters were invalid.
+|===
```

Comment:
can also return SBI_ERR_INVALID_ADDRESS if any of the above address is invalid ?

Reply:
Possibly, don't believe we return that code in our implementation, but will update.

Reply:
Not resovled?